inst/include/datatableAPI.h

```diff
@@ -0,0 +1,45 @@
+
+/* This header file provides the interface used by other packages,
+   and should be included once per package.                        */
+
+#ifndef _R_data_table_API_h_
+#define _R_data_table_API_h_
+
+/* number of R header files (possibly listing too many) */
+#include <R.h>
+
+#ifdef HAVE_VISIBILITY_ATTRIBUTE
+    # define attribute_hidden __attribute__ ((visibility ("hidden")))
+#else
+    # define attribute_hidden
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* provided the interface for the function exported in
+   ../src/init.c via R_RegisterCCallable()		*/
+
+SEXP attribute_hidden DT_subsetDT(SEXP x, SEXP rows, SEXP cols) {
+     static SEXP(*fun)(SEXP, SEXP, SEXP) =
+       (SEXP(*)(SEXP,SEXP,SEXP)) R_GetCCallable("data.table", "CsubsetDT");
+     return fun(x,rows,cols);
```

Comment:
is it necessary to create `fun` object here? couldn't this be a one liner to returning `R_GetCCallable()` call?

Reply:
Yes: `fun` is what is invoked when you call.  The `static` makes the assignment stick, and the `R_GetCCallable()` ges the identifier from the _other_ package.